File: sdks/swift/HookSniffSDK/Sources/HookSniff/AnalyticsResource.swift
import Foundation

/// Analytics resource — delivery stats and trends.
public final class AnalyticsResource {
    private let client: HookSniff

    init(client: HookSniff) {
        self.client = client
    }

    /// Get delivery trends.
    public func deliveryTrends(endpointId: String? = nil) async throws -> [String: Any] {
        var path = "/v1/analytics/delivery-trends"
        if let id = endpointId { path += "?endpoint_id=\(id)" }
        let body = try await client.requestJSON(method: "GET", path: path)
        return JSONHelpers.dict(from: body)
    }

    /// Get latency trends.
    public func latencyTrends(endpointId: String? = nil) async throws -> [String: Any] {
        var path = "/v1/analytics/latency-trends"
        if let id = endpointId { path += "?endpoint_id=\(id)" }
        let body = try await client.requestJSON(method: "GET", path: path)
        return JSONHelpers.dict(from: body)
    }

    /// Get success rate.
    public func successRate(endpointId: String? = nil) async throws -> [String: Any] {
        var path = "/v1/analytics/success-rate"
        if let id = endpointId { path += "?endpoint_id=\(id)" }
        let body = try await client.requestJSON(method: "GET", path: path)
        return JSONHelpers.dict(from: body)
    }
}

File: sdks/kotlin/src/main/kotlin/hooksniff/sdk/wrapper/AnalyticsResource.kt
package hooksniff.sdk.wrapper

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

/**
 * Delivery analytics and statistics.
 */
class AnalyticsResource(private val client: HookSniff) {

    private val gson = Gson()
    private val mapType = object : TypeToken<Map<String, Any?>>() {}.type

    /** Get delivery statistics. */
    fun stats(): Map<String, Any?> {
        val json = client.get("/v1/analytics/stats")
        return gson.fromJson(json, mapType)
    }

    /** Get delivery trends. */
    fun trends(): Map<String, Any?> {
        val json = client.get("/v1/analytics/trends")
        return gson.fromJson(json, mapType)
    }

    /** Get success rate. */
    fun successRate(): Map<String, Any?> {
        val json = client.get("/v1/analytics/success-rate")
        return gson.fromJson(json, mapType)
    }

    /** Get latency data. */
    fun latency(): Map<String, Any?> {
        val json = client.get("/v1/analytics/latency")
        return gson.fromJson(json, mapType)
    }
}
